hack/util.sh

```diff
@@ -660,7 +660,19 @@ function install_registry() {
 readonly -f install_registry
 
 function wait_for_registry() {
-	wait_for_command "oc get endpoints docker-registry --template='{{ len .subsets }}' --config='${ADMIN_KUBECONFIG}' | grep -q '[1-9][0-9]*'" $((5*TIME_MIN))
+	local latestversion=$(oc get dc/docker-registry -o jsonpath='{.status.latestVersion}' --config="${ADMIN_KUBECONFIG}")
+	if [[ -z "${latestversion:-}" ]]; then
+		echo "Failed to get latest version of docker-registry deployment config"
+		exit 1
+	fi
+	wait_for_command "oc get rc/docker-registry-$latestversion --template \"{{ index .metadata.annotations \\\"openshift.io/deployment.phase\\\" }}\" --config='${ADMIN_KUBECONFIG}' | grep Complete"  $((5*TIME_MIN))
+	local registrypod=$(oc get pod -l "deployment=docker-registry-$latestversion" -o jsonpath=$'{.items[*].metadata.name}\n' --config="${ADMIN_KUBECONFIG}" | head -n 1)
+	if [[ -z "${registrypod:-}" ]]; then
+		echo "Failed to get docker registry pod name"
+		exit 1
+	fi
+	local podnamejs='{range .subsets[*]}{range .addresses[*]}{.targetRef.name},{end}{end}'
+	wait_for_command "oc get endpoints/docker-registry -o 'jsonpath=$podnamejs' --config='${ADMIN_KUBECONFIG}' | egrep -q '(^|,)${registrypod},'" $TIME_MIN
```

Comment:
> why is this necessary?

Not sure, I was just reluctant to remove the existing guards.

In my local test, this passes in no time:

```
[INFO] Success running command: 'oc get rc/docker-registry-1 --template "{{ index .metadata.annotations \"openshift.io/deployment.phase\" }}" --config='/tmp/openshift/test-end-to-end//openshift.local.config/master/admin.kubeconfig' | grep Complete' after 23 seconds
[INFO] Waiting for command to finish: 'oc get endpoints/docker-registry -o 'jsonpath={range .subsets[*]}{range .addresses[*]}{.targetRef.name},{end}{end}' --config='/tmp/openshift/test-end-to-end//openshift.local.config/master/admin.kubeconfig' | egrep -q '(^|,)docker-registry-1-6mc18,''...
[INFO] Success running command: 'oc get endpoints/docker-registry -o 'jsonpath={range .subsets[*]}{range .addresses[*]}{.targetRef.name},{end}{end}' --config='/tmp/openshift/test-end-to-end//openshift.local.config/master/admin.kubeconfig' | egrep -q '(^|,)docker-registry-1-6mc18,'' after 0 seconds
[INFO] Waiting for command to finish: 'oc get 'pod/docker-registry-1-6mc18' -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}' --config='/tmp/openshift/test-end-to-end//openshift.local.config/master/admin.kubeconfig' | grep -qi true'...
[INFO] Success running command: 'oc get 'pod/docker-registry-1-6mc18' -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}' --config='/tmp/openshift/test-end-to-end//openshift.local.config/master/admin.kubeconfig' | grep -qi true' after 0 seconds
```